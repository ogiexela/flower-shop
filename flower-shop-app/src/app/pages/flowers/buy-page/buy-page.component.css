.flower-list {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
}

.buy-page-header {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.buy-page-header * {
    margin: 5px;
}

.flower-list-controls {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
}

.flower-list-controls * {
    margin: 10px;
}

button {
    cursor: pointer;
}

.empty-list {
    width: 100%;
    text-align: center;
}
