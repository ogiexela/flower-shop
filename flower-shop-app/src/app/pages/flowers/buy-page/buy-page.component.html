<article>
    <div *ngIf="flowers?.length" class="buy-page-header">
        <div class="cart">
            <label>Cart:</label>
            <select
                name="cars"
                id="cars"
            >
                <option *ngIf="!cart?.length">empty</option>
                <option
                    *ngFor="let item of cart"
                    [value]="item"
                >{{getFlowerName(item)}}</option>
            </select>
        </div>
        <div class="items-per-page">
            <label for="itemsPerPage">Items per page: </label>
            <select
                class="form-control"
                id="itemsPerPage"
                required
                [(ngModel)]="itemsPerPage"
                (ngModelChange)="changeItemsPerPage($event)"
                name="itemsPerPage"
                #power="ngModel"
            >
                <option value="1">1</option>
                <option value="5">5</option>
                <option value="10">10</option>
                <option value="20">20</option>
            </select>
        </div>
    </div>
    <div class="flower-list">
        <app-flower
            (addToCart)="addToCart($event)"
            (removeFromCart)="removeFromCart($event)"
            *ngFor="let flower of flowers;"
            [flower]="flower"
            [inCart]="isInCart(flower.id)"
        ></app-flower>
    </div>
    <div
        *ngIf="flowers?.length"
        class="flower-list-controls"
    >
        <button
            (click)="previousPage()"
            [disabled]="!shouldShowPreviousPageButton"
            id="previous_page"
        >PREVIOUS</button>
        <span class="page-counter">{{currentPage}} of {{flowersResponse.pageCount}}</span>
        <button
            (click)="nextPage()"
            [disabled]="!shouldShowNextPageButton"
            id="next_page"
        >NEXT</button>
    </div>
    <div
        class="empty-list"
        *ngIf="!flowers?.length"
    >
        <p>No flowers in stock.</p>
        <a routerLink="/flowers/sell">Try selling some...</a>
    </div>
</article>
