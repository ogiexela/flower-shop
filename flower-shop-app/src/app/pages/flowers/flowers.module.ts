import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FlowerModule } from '../../components/flower/flower.module';
import { FlowersRoutingModule } from './flowers-routing.module';

@NgModule({
  imports: [
    CommonModule,
    FlowerModule,
    FlowersRoutingModule,
  ]
})
export class FlowersModule { }
