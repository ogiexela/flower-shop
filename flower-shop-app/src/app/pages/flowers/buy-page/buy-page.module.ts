import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { FlowerModule } from '../../../components/flower/flower.module';
import { BuyPageComponent } from './buy-page.component';

@NgModule({
  declarations: [
    BuyPageComponent,
  ],
  imports: [
    CommonModule,
    FlowerModule,
    FormsModule,
    RouterModule,
  ]
})
export class BuyPageModule { }
