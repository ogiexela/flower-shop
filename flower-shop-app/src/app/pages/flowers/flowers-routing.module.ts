import { NgModule } from '@angular/core';
import {
  RouterModule,
  Routes,
} from '@angular/router';
import { FLOWER_ROUTE } from '../../constants/routes/flower.routes';
import { BuyPageComponent } from './buy-page/buy-page.component';
import { BuyPageModule } from './buy-page/buy-page.module';
import { SellPageComponent } from './sell-page/sell-page.component';
import { SellPageModule } from './sell-page/sell-page.module';

const routes: Routes = [
  {
    path: FLOWER_ROUTE.BUY,
    component: BuyPageComponent,
  },
  {
    path: FLOWER_ROUTE.SELL,
    component: SellPageComponent
  },
  {
    path: '',
    redirectTo: FLOWER_ROUTE.BUY,
    pathMatch: 'full',
  },
];

@NgModule({
  imports: [
    BuyPageModule,
    SellPageModule,
    RouterModule.forChild(routes),
  ],
  exports: [RouterModule]
})
export class FlowersRoutingModule { }
