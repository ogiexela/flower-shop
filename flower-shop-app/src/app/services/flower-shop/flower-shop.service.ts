import { HttpClient } from '@angular/common/http';
import {
  Inject,
  Injectable,
} from '@angular/core';
import { Flower } from '../../models/flower.model';
import { Flowers } from '../../models/flowers.model';

@Injectable({
  providedIn: 'root'
})
export class FlowerShopService {

  private readonly API_URL = '/api';

  constructor(@Inject(HttpClient) private http: HttpClient) { }

  public getFlowers(page: number = 1, offset: number = 0, limit: number = 25) {
    return this.http.get<Flowers>(`${this.API_URL}/flower?limit=${limit}&offset=${offset}&page=${page}`);
  }

  public createFlower(flower: Flower) {
    return this.http.post<Flower>(`${this.API_URL}/flower`, flower);
  }
}
