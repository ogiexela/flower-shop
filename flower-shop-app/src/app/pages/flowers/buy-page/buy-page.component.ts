import { take } from 'rxjs/operators';
import {
  Component,
  Inject,
  OnInit,
} from '@angular/core';
import { Flower } from '../../../models/flower.model';
import { Flowers } from '../../../models/flowers.model';
import { FlowerShopService } from '../../../services/flower-shop/flower-shop.service';

@Component({
  selector: 'app-buy-page',
  styleUrls: ['./buy-page.component.css'],
  templateUrl: './buy-page.component.html',
})
export class BuyPageComponent implements OnInit {

  public cart = [];
  public allFlowers: Set<Flower> = new Set<Flower>();
  public flowers: Flower[] = [];
  public flowersResponse: Flowers;
  public currentPage = 1;
  public itemsPerPage = 10;

  constructor(
    @Inject(FlowerShopService) private readonly flowerShopService: FlowerShopService,
  ) { }

  ngOnInit() {
    this.getFlowers();
  }

  public nextPage(): void {
    this.currentPage += 1;
    this.getFlowers();
  }

  public previousPage(): void {
    this.currentPage -= 1;
    this.getFlowers();
  }

  public changeItemsPerPage(count: number): void {
    this.itemsPerPage = count;
    this.currentPage = 1;
    this.getFlowers();
  }

  private getFlowers(): void {
    this.flowerShopService
      .getFlowers(+this.currentPage, 0, +this.itemsPerPage)
      .pipe(take(1))
      .subscribe((flowers: Flowers) => {
        this.flowersResponse = flowers;
        this.flowers = flowers.data;
        this.flowers.forEach((flower) => this.allFlowers.add(flower));
      });
  }

  public isInCart(id: string): boolean {
    return this.cart.includes(id);
  }

  public get shouldShowPreviousPageButton(): boolean {
    return this.currentPage > 1;
  }

  public get shouldShowNextPageButton(): boolean {
    return this.currentPage < this.flowersResponse.pageCount;
  }

  public getFlowerName(flowerId: string): string {
    return [...this.allFlowers].find((flower) => flower.id === flowerId).name;
  }

  public addToCart(flowerId: string): void {
    this.cart = this.cart.concat([flowerId]);
  }

  public removeFromCart(flowerId: string): void {
    this.cart = this.cart.filter((id) => id !== flowerId);
  }

}
