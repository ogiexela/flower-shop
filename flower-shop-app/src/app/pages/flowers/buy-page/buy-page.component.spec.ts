import { HttpClientTestingModule } from '@angular/common/http/testing';
import {
  async,
  ComponentFixture,
  getTestBed,
  TestBed,
} from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { RouterTestingModule } from '@angular/router/testing';
import { Flowers } from '../../../models/flowers.model';
import { FlowerShopService } from '../../../services/flower-shop/flower-shop.service';
import { BuyPageComponent } from './buy-page.component';
import { BuyPageModule } from './buy-page.module';

const testData: Flowers = {
  data: [],
  count: 0,
  pageCount: 0,
  total: 0,
  page: 0,
};

class MockFlowerShopServiceService {
  public getFlowers() {
    return {
      pipe: () => {
        return {
          subscribe: (next, error) => {
            next(testData);
            if (error) {
              error();
            }
          }
        };
      }
    };
  }
}

describe('BuyPageComponent', () => {

  let component: BuyPageComponent;
  let fixture: ComponentFixture<BuyPageComponent>;

  beforeEach(async(() => {
    window.alert = jest.fn();

    TestBed.configureTestingModule({
      declarations: [],
      imports: [BuyPageModule, HttpClientTestingModule, RouterTestingModule.withRoutes([])],
      providers: [
        HttpClientTestingModule,
        RouterTestingModule,
        { provide: FlowerShopService, useClass: MockFlowerShopServiceService }
      ]
    })
      .compileComponents()
      .then(() => {
        fixture = TestBed.createComponent(BuyPageComponent);
        fixture.detectChanges();
        component = fixture.componentInstance;
      })
      .catch(error => console.error(error));
  }));

  it('empty state', () => {
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot('empty_state');
  });

  it('one page total', async () => {
    const testData1: Flowers = {
      data: [{
        name: 'test flower1',
        sku: 'f1',
        id: 'foooid',
        image: 'foo.jpg',
        price: 1.98,
        rating: 2,
      }],
      count: 0,
      pageCount: 0,
      total: 0,
      page: 0,
    };

    const flowerShopService: FlowerShopService = getTestBed().get(FlowerShopService);

    flowerShopService.getFlowers = jest.fn().mockImplementation(() => {
      return {
        pipe: () => {
          return {
            subscribe: (next, error) => {
              next(testData1);
              if (error) {
                error();
              }
            }
          };
        }
      };
    });

    fixture.componentInstance.ngOnInit();

    fixture.detectChanges();

    await fixture.whenStable();

    expect(fixture.nativeElement).toMatchSnapshot('one_flower_state');
  });

  it('add flower to cart', async () => {
    const testData1: Flowers = {
      data: [{
        name: 'test flower1',
        sku: 'f1',
        id: 'foooid',
        image: 'foo.jpg',
        price: 1.98,
        rating: 2,
      }],
      count: 0,
      pageCount: 0,
      total: 0,
      page: 0,
    };

    const flowerShopService: FlowerShopService = getTestBed().get(FlowerShopService);

    flowerShopService.getFlowers = jest.fn().mockImplementation(() => {
      return {
        pipe: () => {
          return {
            subscribe: (next, error) => {
              next(testData1);
              if (error) {
                error();
              }
            }
          };
        }
      };
    });

    fixture.componentInstance.ngOnInit();

    fixture.detectChanges();

    const firstFlower = fixture.debugElement.query(By.css(`#flower_card_${testData1.data[0].id}`));
    const mouseenter = new MouseEvent('mouseenter');
    firstFlower.nativeElement.dispatchEvent(mouseenter);

    fixture.detectChanges();

    const addToCart = firstFlower.query(By.css('#add_flower_to_cart'));
    const click = new MouseEvent('click');

    addToCart.nativeElement.dispatchEvent(click);

    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot('flower_in_cart_state');
  });

  it('remove flower from cart', async () => {
    const testData1: Flowers = {
      data: [{
        name: 'test flower1',
        sku: 'f1',
        id: 'foooid',
        image: 'foo.jpg',
        price: 1.98,
        rating: 2,
      }],
      count: 0,
      pageCount: 0,
      total: 0,
      page: 0,
    };

    const flowerShopService: FlowerShopService = getTestBed().get(FlowerShopService);

    flowerShopService.getFlowers = jest.fn().mockImplementation(() => {
      return {
        pipe: () => {
          return {
            subscribe: (next, error) => {
              next(testData1);
              if (error) {
                error();
              }
            }
          };
        }
      };
    });

    fixture.componentInstance.ngOnInit();

    fixture.detectChanges();

    fixture.componentInstance.cart = [testData1.data[0].id];

    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot('flower_in_cart_state');

    const firstFlower = fixture.debugElement.query(By.css(`#flower_card_${testData1.data[0].id}`));

    const mouseenter = new MouseEvent('mouseenter');
    firstFlower.nativeElement.dispatchEvent(mouseenter);

    fixture.detectChanges();

    const removeFromCart = firstFlower.query(By.css('#remove_flower_from_cart'));
    const click = new MouseEvent('click');

    removeFromCart.nativeElement.dispatchEvent(click);

    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot('cart_empty_state');
  });

  it('next and previous pages of flowers', async () => {
    const testData1: Flowers = {
      data: [
        {
          name: 'test flower1',
          sku: 'f1',
          id: 'foooid1',
          image: 'foo1.jpg',
          price: 1.98,
          rating: 2,
        },
        {
          name: 'test flower2',
          sku: 'f2',
          id: 'foooid2',
          image: 'foo2.jpg',
          price: 2.22,
          rating: 4,
        },
      ],
      count: 1,
      pageCount: 2,
      total: 2,
      page: 1,
    };

    const testData2: Flowers = {
      data: [
        {
          name: 'test flower2',
          sku: 'f2',
          id: 'foooid2',
          image: 'foo2.jpg',
          price: 2.22,
          rating: 4,
        },
      ],
      count: 1,
      pageCount: 2,
      total: 2,
      page: 2,
    };

    const flowerShopService: FlowerShopService = getTestBed().get(FlowerShopService);

    flowerShopService.getFlowers = jest.fn().mockImplementation(() => {
      return {
        pipe: () => {
          return {
            subscribe: (next, error) => {
              next(testData1);
              if (error) {
                error();
              }
            }
          };
        }
      };
    });

    fixture.componentInstance.ngOnInit();

    fixture.detectChanges();

    fixture.componentInstance.currentPage = 1;
    fixture.componentInstance.itemsPerPage = 1;

    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot('first_page');

    flowerShopService.getFlowers = jest.fn().mockImplementation(() => {
      return {
        pipe: () => {
          return {
            subscribe: (next, error) => {
              next(testData2);
              if (error) {
                error();
              }
            }
          };
        }
      };
    });

    const nextPageButton = fixture.debugElement.query(By.css('#next_page'));
    const click = new MouseEvent('click');

    nextPageButton.nativeElement.dispatchEvent(click);

    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot('next_page');

    flowerShopService.getFlowers = jest.fn().mockImplementation(() => {
      return {
        pipe: () => {
          return {
            subscribe: (next, error) => {
              next(testData1);
              if (error) {
                error();
              }
            }
          };
        }
      };
    });

    const previousPageButton = fixture.debugElement.query(By.css('#previous_page'));

    previousPageButton.nativeElement.dispatchEvent(click);

    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot('previous_page');
  });


  it('change items per page', async () => {
    const testData1: Flowers = {
      data: [
        {
          name: 'test flower1',
          sku: 'f1',
          id: 'foooid1',
          image: 'foo1.jpg',
          price: 1.98,
          rating: 2,
        },
        {
          name: 'test flower2',
          sku: 'f2',
          id: 'foooid2',
          image: 'foo2.jpg',
          price: 2.22,
          rating: 4,
        },
      ],
      count: 1,
      pageCount: 2,
      total: 2,
      page: 1,
    };

    const flowerShopService: FlowerShopService = getTestBed().get(FlowerShopService);

    flowerShopService.getFlowers = jest.fn().mockImplementation(() => {
      return {
        pipe: () => {
          return {
            subscribe: (next, error) => {
              next(testData1);
              if (error) {
                error();
              }
            }
          };
        }
      };
    });

    fixture.componentInstance.ngOnInit();

    fixture.detectChanges();

    const itemsPerPage: HTMLSelectElement = fixture.debugElement.query(By.css('#itemsPerPage')).nativeElement;
    itemsPerPage.selectedIndex = 1;

    itemsPerPage.dispatchEvent(new Event('change'));

    fixture.detectChanges();

    await fixture.whenStable();

    expect(flowerShopService.getFlowers).toHaveBeenCalledWith(1, 0, 5);
  });
});
