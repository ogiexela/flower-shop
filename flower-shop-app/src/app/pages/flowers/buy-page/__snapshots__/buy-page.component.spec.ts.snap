// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyPageComponent add flower to cart: flower_in_cart_state 1`] = `
<div
  id="root2"
>
  <article>
    
    <div
      class="buy-page-header"
    >
      <div
        class="cart"
      >
        <label>
          Cart:
        </label>
        <select
          id="cars"
          name="cars"
        >
          
          
          <option
            value="foooid"
          >
            test flower1
          </option>
        </select>
      </div>
      <div
        class="items-per-page"
      >
        <label
          for="itemsPerPage"
        >
          Items per page: 
        </label>
        <select
          class="form-control ng-untouched ng-pristine ng-valid"
          id="itemsPerPage"
          name="itemsPerPage"
          required=""
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="20"
          >
            20
          </option>
        </select>
      </div>
    </div>
    <div
      class="flower-list"
    >
      
      <app-flower>
        
        <div
          class="flower-card"
          id="flower_card_foooid"
        >
          <div
            class="flower-image"
          >
            <img
              class="half-opacity"
              src="./assets/flowers/foo.jpg"
            />
            
            
            <button
              id="remove_flower_from_cart"
            >
              Remove from cart
            </button>
          </div>
          <span
            class="flower-name"
          >
            test flower1
          </span>
          <span
            class="flower-price"
          >
            $1.98
          </span>
          <div
            class="flower-rating"
          >
            
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ☆
            </span>
            <span>
              ☆
            </span>
            <span>
              ☆
            </span>
          </div>
          
          <div
            class="in-cart"
          >
            <span>
              In Cart
            </span>
          </div>
        </div>
        
      </app-flower>
    </div>
    
    <div
      class="flower-list-controls"
    >
      <button
        disabled=""
        id="previous_page"
      >
        PREVIOUS
      </button>
      <span
        class="page-counter"
      >
        1 of 0
      </span>
      <button
        disabled=""
        id="next_page"
      >
        NEXT
      </button>
    </div>
    
  </article>
</div>
`;

exports[`BuyPageComponent empty state: empty_state 1`] = `
<div
  id="root0"
>
  <article>
    
    <div
      class="flower-list"
    >
      
    </div>
    
    
    <div
      class="empty-list"
    >
      <p>
        No flowers in stock.
      </p>
      <a
        href="/flowers/sell"
        routerlink="/flowers/sell"
      >
        Try selling some...
      </a>
    </div>
  </article>
</div>
`;

exports[`BuyPageComponent next and previous pages of flowers: first_page 1`] = `
<div
  id="root4"
>
  <article>
    
    <div
      class="buy-page-header"
    >
      <div
        class="cart"
      >
        <label>
          Cart:
        </label>
        <select
          id="cars"
          name="cars"
        >
          
          <option>
            empty
          </option>
          
        </select>
      </div>
      <div
        class="items-per-page"
      >
        <label
          for="itemsPerPage"
        >
          Items per page: 
        </label>
        <select
          class="form-control ng-untouched ng-pristine ng-invalid"
          id="itemsPerPage"
          name="itemsPerPage"
          required=""
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="20"
          >
            20
          </option>
        </select>
      </div>
    </div>
    <div
      class="flower-list"
    >
      
      <app-flower>
        
        <div
          class="flower-card"
          id="flower_card_foooid1"
        >
          <div
            class="flower-image"
          >
            <img
              src="./assets/flowers/foo1.jpg"
            />
            
            
          </div>
          <span
            class="flower-name"
          >
            test flower1
          </span>
          <span
            class="flower-price"
          >
            $1.98
          </span>
          <div
            class="flower-rating"
          >
            
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ☆
            </span>
            <span>
              ☆
            </span>
            <span>
              ☆
            </span>
          </div>
          
        </div>
        
      </app-flower>
      <app-flower>
        
        <div
          class="flower-card"
          id="flower_card_foooid2"
        >
          <div
            class="flower-image"
          >
            <img
              src="./assets/flowers/foo2.jpg"
            />
            
            
          </div>
          <span
            class="flower-name"
          >
            test flower2
          </span>
          <span
            class="flower-price"
          >
            $2.22
          </span>
          <div
            class="flower-rating"
          >
            
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ☆
            </span>
          </div>
          
        </div>
        
      </app-flower>
    </div>
    
    <div
      class="flower-list-controls"
    >
      <button
        disabled=""
        id="previous_page"
      >
        PREVIOUS
      </button>
      <span
        class="page-counter"
      >
        1 of 2
      </span>
      <button
        id="next_page"
      >
        NEXT
      </button>
    </div>
    
  </article>
</div>
`;

exports[`BuyPageComponent next and previous pages of flowers: next_page 1`] = `
<div
  id="root4"
>
  <article>
    
    <div
      class="buy-page-header"
    >
      <div
        class="cart"
      >
        <label>
          Cart:
        </label>
        <select
          id="cars"
          name="cars"
        >
          
          <option>
            empty
          </option>
          
        </select>
      </div>
      <div
        class="items-per-page"
      >
        <label
          for="itemsPerPage"
        >
          Items per page: 
        </label>
        <select
          class="form-control ng-untouched ng-pristine ng-valid"
          id="itemsPerPage"
          name="itemsPerPage"
          required=""
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="20"
          >
            20
          </option>
        </select>
      </div>
    </div>
    <div
      class="flower-list"
    >
      
      <app-flower>
        
        <div
          class="flower-card"
          id="flower_card_foooid2"
        >
          <div
            class="flower-image"
          >
            <img
              src="./assets/flowers/foo2.jpg"
            />
            
            
          </div>
          <span
            class="flower-name"
          >
            test flower2
          </span>
          <span
            class="flower-price"
          >
            $2.22
          </span>
          <div
            class="flower-rating"
          >
            
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ☆
            </span>
          </div>
          
        </div>
        
      </app-flower>
    </div>
    
    <div
      class="flower-list-controls"
    >
      <button
        id="previous_page"
      >
        PREVIOUS
      </button>
      <span
        class="page-counter"
      >
        2 of 2
      </span>
      <button
        disabled=""
        id="next_page"
      >
        NEXT
      </button>
    </div>
    
  </article>
</div>
`;

exports[`BuyPageComponent next and previous pages of flowers: previous_page 1`] = `
<div
  id="root4"
>
  <article>
    
    <div
      class="buy-page-header"
    >
      <div
        class="cart"
      >
        <label>
          Cart:
        </label>
        <select
          id="cars"
          name="cars"
        >
          
          <option>
            empty
          </option>
          
        </select>
      </div>
      <div
        class="items-per-page"
      >
        <label
          for="itemsPerPage"
        >
          Items per page: 
        </label>
        <select
          class="form-control ng-untouched ng-pristine ng-valid"
          id="itemsPerPage"
          name="itemsPerPage"
          required=""
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="20"
          >
            20
          </option>
        </select>
      </div>
    </div>
    <div
      class="flower-list"
    >
      
      <app-flower>
        
        <div
          class="flower-card"
          id="flower_card_foooid1"
        >
          <div
            class="flower-image"
          >
            <img
              src="./assets/flowers/foo1.jpg"
            />
            
            
          </div>
          <span
            class="flower-name"
          >
            test flower1
          </span>
          <span
            class="flower-price"
          >
            $1.98
          </span>
          <div
            class="flower-rating"
          >
            
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ☆
            </span>
            <span>
              ☆
            </span>
            <span>
              ☆
            </span>
          </div>
          
        </div>
        
      </app-flower>
      <app-flower>
        
        <div
          class="flower-card"
          id="flower_card_foooid2"
        >
          <div
            class="flower-image"
          >
            <img
              src="./assets/flowers/foo2.jpg"
            />
            
            
          </div>
          <span
            class="flower-name"
          >
            test flower2
          </span>
          <span
            class="flower-price"
          >
            $2.22
          </span>
          <div
            class="flower-rating"
          >
            
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ☆
            </span>
          </div>
          
        </div>
        
      </app-flower>
    </div>
    
    <div
      class="flower-list-controls"
    >
      <button
        disabled=""
        id="previous_page"
      >
        PREVIOUS
      </button>
      <span
        class="page-counter"
      >
        1 of 2
      </span>
      <button
        id="next_page"
      >
        NEXT
      </button>
    </div>
    
  </article>
</div>
`;

exports[`BuyPageComponent one page total: one_flower_state 1`] = `
<div
  id="root1"
>
  <article>
    
    <div
      class="buy-page-header"
    >
      <div
        class="cart"
      >
        <label>
          Cart:
        </label>
        <select
          id="cars"
          name="cars"
        >
          
          <option>
            empty
          </option>
          
        </select>
      </div>
      <div
        class="items-per-page"
      >
        <label
          for="itemsPerPage"
        >
          Items per page: 
        </label>
        <select
          class="form-control ng-untouched ng-pristine ng-invalid"
          id="itemsPerPage"
          name="itemsPerPage"
          required=""
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="20"
          >
            20
          </option>
        </select>
      </div>
    </div>
    <div
      class="flower-list"
    >
      
      <app-flower>
        
        <div
          class="flower-card"
          id="flower_card_foooid"
        >
          <div
            class="flower-image"
          >
            <img
              src="./assets/flowers/foo.jpg"
            />
            
            
          </div>
          <span
            class="flower-name"
          >
            test flower1
          </span>
          <span
            class="flower-price"
          >
            $1.98
          </span>
          <div
            class="flower-rating"
          >
            
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ☆
            </span>
            <span>
              ☆
            </span>
            <span>
              ☆
            </span>
          </div>
          
        </div>
        
      </app-flower>
    </div>
    
    <div
      class="flower-list-controls"
    >
      <button
        disabled=""
        id="previous_page"
      >
        PREVIOUS
      </button>
      <span
        class="page-counter"
      >
        1 of 0
      </span>
      <button
        disabled=""
        id="next_page"
      >
        NEXT
      </button>
    </div>
    
  </article>
</div>
`;

exports[`BuyPageComponent remove flower from cart: cart_empty_state 1`] = `
<div
  id="root3"
>
  <article>
    
    <div
      class="buy-page-header"
    >
      <div
        class="cart"
      >
        <label>
          Cart:
        </label>
        <select
          id="cars"
          name="cars"
        >
          
          <option>
            empty
          </option>
          
        </select>
      </div>
      <div
        class="items-per-page"
      >
        <label
          for="itemsPerPage"
        >
          Items per page: 
        </label>
        <select
          class="form-control ng-untouched ng-pristine ng-valid"
          id="itemsPerPage"
          name="itemsPerPage"
          required=""
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="20"
          >
            20
          </option>
        </select>
      </div>
    </div>
    <div
      class="flower-list"
    >
      
      <app-flower>
        
        <div
          class="flower-card"
          id="flower_card_foooid"
        >
          <div
            class="flower-image"
          >
            <img
              class=""
              src="./assets/flowers/foo.jpg"
            />
            
            
          </div>
          <span
            class="flower-name"
          >
            test flower1
          </span>
          <span
            class="flower-price"
          >
            $1.98
          </span>
          <div
            class="flower-rating"
          >
            
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ☆
            </span>
            <span>
              ☆
            </span>
            <span>
              ☆
            </span>
          </div>
          
        </div>
        
      </app-flower>
    </div>
    
    <div
      class="flower-list-controls"
    >
      <button
        disabled=""
        id="previous_page"
      >
        PREVIOUS
      </button>
      <span
        class="page-counter"
      >
        1 of 0
      </span>
      <button
        disabled=""
        id="next_page"
      >
        NEXT
      </button>
    </div>
    
  </article>
</div>
`;

exports[`BuyPageComponent remove flower from cart: flower_in_cart_state 1`] = `
<div
  id="root3"
>
  <article>
    
    <div
      class="buy-page-header"
    >
      <div
        class="cart"
      >
        <label>
          Cart:
        </label>
        <select
          id="cars"
          name="cars"
        >
          
          
          <option
            value="foooid"
          >
            test flower1
          </option>
        </select>
      </div>
      <div
        class="items-per-page"
      >
        <label
          for="itemsPerPage"
        >
          Items per page: 
        </label>
        <select
          class="form-control ng-untouched ng-pristine ng-invalid"
          id="itemsPerPage"
          name="itemsPerPage"
          required=""
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="20"
          >
            20
          </option>
        </select>
      </div>
    </div>
    <div
      class="flower-list"
    >
      
      <app-flower>
        
        <div
          class="flower-card"
          id="flower_card_foooid"
        >
          <div
            class="flower-image"
          >
            <img
              class="half-opacity"
              src="./assets/flowers/foo.jpg"
            />
            
            
            <button
              id="remove_flower_from_cart"
            >
              Remove from cart
            </button>
          </div>
          <span
            class="flower-name"
          >
            test flower1
          </span>
          <span
            class="flower-price"
          >
            $1.98
          </span>
          <div
            class="flower-rating"
          >
            
            <span>
              ★
            </span>
            <span>
              ★
            </span>
            <span>
              ☆
            </span>
            <span>
              ☆
            </span>
            <span>
              ☆
            </span>
          </div>
          
          <div
            class="in-cart"
          >
            <span>
              In Cart
            </span>
          </div>
        </div>
        
      </app-flower>
    </div>
    
    <div
      class="flower-list-controls"
    >
      <button
        disabled=""
        id="previous_page"
      >
        PREVIOUS
      </button>
      <span
        class="page-counter"
      >
        1 of 0
      </span>
      <button
        disabled=""
        id="next_page"
      >
        NEXT
      </button>
    </div>
    
  </article>
</div>
`;
